bot-api/EducationalTeamsBotApi/EducationalTeamsBotApi/EducationalTeamsBotApi.Application/Speakers/Commands/EditSpeakerCommand/UpsertSpeakerCommandHandler.cs

```diff
@@ -12,28 +12,36 @@ namespace EducationalTeamsBotApi.Application.Speakers.Commands.EditSpeakerComman
     using MediatR;
 
     /// <summary>
-    /// Command handler of the speaker edition.
+    /// Command handler of the speaker insertion and edition.
     /// </summary>
-    public class EditSpeakerCommandHandler : IRequestHandler<EditSpeakerCommand, CosmosSpeaker?>
+    public class UpsertSpeakerCommandHandler : IRequestHandler<UpsertSpeakerCommand, CosmosSpeaker?>
     {
         /// <summary>
         /// Speaker service.
         /// </summary>
         private readonly ISpeakerCosmosService speakerService;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="EditSpeakerCommandHandler"/> class.
+        /// Initializes a new instance of the <see cref="UpsertSpeakerCommandHandler"/> class.
         /// </summary>
         /// <param name="speakerService">Service of the speakers.</param>
-        public EditSpeakerCommandHandler(ISpeakerCosmosService speakerService)
+        public UpsertSpeakerCommandHandler(ISpeakerCosmosService speakerService)
         {
             this.speakerService = speakerService;
         }
 
         /// <inheritdoc/>
-        public async Task<CosmosSpeaker?> Handle(EditSpeakerCommand request, CancellationToken cancellationToken)
+        public async Task<CosmosSpeaker?> Handle(UpsertSpeakerCommand request, CancellationToken cancellationToken)
         {
-            return await this.speakerService.EditSpeaker(request.Speaker);
+            var speaker = this.speakerService.GetSpeaker(request.Speaker.Id);
+            if (speaker.Result == null)
+            {
+                return await this.speakerService.AddSpeaker(request.Speaker);
+            }
+            else
+            {
+                return await this.speakerService.EditSpeaker(request.Speaker);
+            }
```

Comment:
`return speaker.Result ? await this.speakerService.AddSpeaker(request.Speaker) : this.speakerService.EditSpeaker(request.Speaker) `

Vraiment pas obligatoire mais je trouve dommage de faire 8 lignes pour un truc réalisable en une